common/beerocks/bwl/include/bwl/nl80211_client.h

```diff
@@ -33,6 +51,116 @@ namespace bwl {
+         * Possible values are 0 (unknown), 20, 40, 80 and 160 MHz.
+         */
+        uint16_t bandwidth_MHz = 0;
+
+        /**
+         * Maximum transmission power in dBm.
+         * Obtained as 0.01 * NL80211_FREQUENCY_ATTR_MAX_TX_POWER
+         */
+        uint32_t tx_pow_dBm = 0;
+
+        /**
+         * Radar detection is mandatory on this channel in current regulatory domain.
+         * Set to true if attribute NL80211_FREQUENCY_ATTR_RADAR is present.
+         */
+        bool is_dfs = false;
+
+        /**
+         * True if current state for DFS is NL80211_DFS_UNAVAILABLE.
+         * Obtained from NL80211_FREQUENCY_ATTR_DFS_STATE
+         */
+        bool radar_affected = false;
+    };
+
+    /**
+     * @brief Band information
+     *
+     * Used in sRadioInfo structure as each radio contains a list of bands.
+     *
+     * Information obtained with NL80211_CMD_GET_WIPHY command through a NL80211 socket.
+     * See NL80211_BAND_ATTR_* in <linux/nl80211.h> for a description of each field.
+     */
+    struct sBandInfo {
+
+        /**
+         * Value of NL80211_BAND_ATTR_HT_CAPA, see iw/util.c print_ht_capability() as a bit
+         * interpretation example.
+         */
+        uint16_t ht_capability = 0;
+
+        /**
+         * Value of NL80211_BAND_ATTR_HT_MCS_SET, see iw/util.c print_ht_mcs() as a byte
+         * interpretation example.
+         */
+        uint8_t ht_mcs_set[ht_mcs_set_size];
+
+        /**
+         * Value of NL80211_BAND_ATTR_VHT_CAPA, see iw/util.c print_vht_info() as a bit
+         * interpretation example.
+         */
+        uint32_t vht_capability = 0;
+
+        /**
+         * Value of NL80211_BAND_ATTR_VHT_MCS_SET, see iw/util.c print_vht_info() as a byte
+         * interpretation example.
+         */
+        uint8_t vht_mcs_set[vht_mcs_set_size];
+
+        /**
+         * Channels supported in this band (obtained from NL80211_BAND_ATTR_FREQS).
+         */
+        std::vector<sChannelInfo> supported_channels;
+
+        /**
+         * Frequency band.
+         * The value for this field can be obtained from whatever of the supported channels. It is
+         * included as a separate field to facilitate its access.
+         */
+        beerocks::eFreqType frequency_band = beerocks::eFreqType::FREQ_UNKNOWN;
+
+        /**
+         * Maximum bandwidth in supported channels.
+         * Value is computed while filling in the list of supported channels.
+         * The value for this field can be obtained from the list of the supported channels. It is
+         * included as a separate field to facilitate its access.
+         */
+        beerocks::eWiFiBandwidth max_bandwidth = beerocks::eWiFiBandwidth::BANDWIDTH_20;
+    };
+
+    /**
+     * @brief Radio information
+     *
+     * Information obtained with NL80211_CMD_GET_WIPHY command through a NL80211 socket.
+     * See <linux/nl80211.h> for a description of each field.
+     */
+    struct sRadioInfo {
```

Comment:
There is already a radio info struct in `base_wlan_hal_types.h` - can we extend it and use it here instead?

Reply:
There is a radio info struct in `base_wlan_hal_types.h` and another one in `beerocks_message_structs.h`. Which one should I use?
The same happens with the WiFi channel struct.

Reply:
Done as agreed during the meeting: use new sRadioInfo structure